oneview/resource_enclosure_group.go

```diff
@@ -191,18 +201,24 @@ func resourceEnclosureGroupCreate(d *schema.ResourceData, meta interface{}) erro
 	enclosureGroup := ov.EnclosureGroup{
 		Name: d.Get("name").(string),
 	}
+	interconnectBayMappings := make([]ov.InterconnectBayMap, 0)
 	if val, ok := d.GetOk("interconnect_bay_mappings"); ok {
 		rawInterconnectBayMappings := val.(*schema.Set).List()
-		interconnectBayMappings := make([]ov.InterconnectBayMap, 0)
 		for _, raw := range rawInterconnectBayMappings {
 			interconnectBayMappingItem := raw.(map[string]interface{})
+			logicalInterconnectGroup, err := config.ovClient.GetLogicalInterconnectGroupByName(interconnectBayMappingItem["logical_interconnect_group_name"].(string))
+			if err != nil || logicalInterconnectGroup.URI.IsNil() {
+				d.SetId("")
+				return nil
```

Comment:
can we return an error statement here please?

Reply:
Done